microsoft/utils/setup_AITL.py

```diff
@@ -0,0 +1,299 @@
+if subscription_id == "":
+    sub_cmd_result = call_cmd('az account list --query "[0].id" --output tsv')
+    subscription_id = "/subscriptions/" + sub_cmd_result.stdout
+    print(
+        f"Input subscription_id is empty, will use 1st default one: {subscription_id} \n"
+    )
+
+action_perms_list = [
+    "Microsoft.Resources/subscriptions/resourceGroups/read",
+    "Microsoft.Resources/subscriptions/resourceGroups/write",
+    "Microsoft.Resources/subscriptions/resourceGroups/delete",
+    "Microsoft.Resources/deployments/read",
+    "Microsoft.Resources/deployments/write",
+    "Microsoft.Resources/deployments/validate/action",
+    "Microsoft.Resources/deployments/operationStatuses/read",
+    "Microsoft.Compute/virtualMachines/read",
+    "Microsoft.Compute/virtualMachines/write",
+    "Microsoft.Compute/virtualMachines/retrieveBootDiagnosticsData/action",
+    "Microsoft.Compute/availabilitySets/write",
+    "Microsoft.Compute/virtualMachines/start/action",
+    "Microsoft.Compute/virtualMachines/restart/action",
+    "Microsoft.Compute/virtualMachines/deallocate/action",
+    "Microsoft.Compute/virtualMachines/powerOff/action",
+    "Microsoft.Compute/disks/read",
+    "Microsoft.Compute/disks/write",
+    "Microsoft.Compute/disks/delete",
+    "Microsoft.Compute/virtualMachines/extensions/read",
+    "Microsoft.Compute/virtualMachines/extensions/write",
+    "Microsoft.Network/virtualNetworks/read",
+    "Microsoft.Network/virtualNetworks/write",
+    "Microsoft.Network/virtualNetworks/subnets/join/action",
+    "Microsoft.Network/publicIPAddresses/read",
+    "Microsoft.Network/publicIPAddresses/write",
+    "Microsoft.Network/publicIPAddresses/join/action",
+    "Microsoft.Network/networkInterfaces/read",
+    "Microsoft.Network/networkInterfaces/write",
+    "Microsoft.Network/networkInterfaces/join/action",
+    "Microsoft.Network/privateEndpoints/write",
+    "Microsoft.Network/privateLinkServices/PrivateEndpointConnectionsApproval/action",
+    "Microsoft.SerialConsole/serialPorts/write",
+    "Microsoft.Network/networkSecurityGroups/write",
+    "Microsoft.Network/networkSecurityGroups/read",
+    "Microsoft.Network/networkSecurityGroups/join/action",
+    "Microsoft.Storage/storageAccounts/read",
+    "Microsoft.Storage/storageAccounts/write",
+    "Microsoft.Storage/storageAccounts/listKeys/action",
+]
+data_action_perms_list = [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/add/action",
+]
+
+target_role = {
+    "Description": "Custmor role is to run test cases and upload logs in Prod env.",
+    "IsCustom": True,
+    "Name": role_name,
+    "Actions": action_perms_list,
+    "DataActions": data_action_perms_list,
+    "AssignableScopes": [subscription_id],
```

Comment:
Does it needed? I'm thinking if users want to enable AITL on multiple subscriptions, this may fail.

Reply:
The "subscription_id" is needed in the scope. The current design: for one customer, if they have different subscriptions, they need to create different roles with different scopes.
 az role list command has some limitation, if doesn't specify "subscription" in scope, then this role can't be found.